proxy/round_tripper/proxy_round_tripper.go

```diff
@@ -163,12 +163,12 @@ func (rt *roundTripper) RoundTrip(originalRequest *http.Request) (*http.Response
 	var selectEndpointErr error
 	var maxAttempts int
 	if reqInfo.RouteServiceURL == nil {
-		maxAttempts = rt.config.Backends.MaxAttempts
+		maxAttempts = max(min(rt.config.Backends.MaxAttempts, reqInfo.RoutePool.NumEndpoints()), 1)
 	} else {
 		maxAttempts = rt.config.RouteServiceConfig.MaxAttempts
 	}
 
-	for attempt := 1; attempt <= maxAttempts || maxAttempts == 0; attempt++ {
+	for attempt := 1; attempt <= maxAttempts; attempt++ {
```

Comment:
This is a breaking change, do we know of anyone relying on this "unlimited" option?

Reply:
yes. see https://github.com/cloudfoundry/routing-release/pull/390

Reply:
Okay but the option was always a bit dangerous to begin with, given that there are retrieable errors which do not cause the endpoint to be pruned 😄